url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/AbstractPosition.java

```diff
@@ -0,0 +1,856 @@
+   {
+      if(colour == Colour.WHITE)
+      {
+         return whiteCastlingShort;
+      }
+      else
+      {
+         return blackCastlingShort;
+      }
+   }
+
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#getEnroqueLargo(es.dagaren.chesi.Colour)
+    */
+   public boolean getCastlingLong(Colour colour)
+   {
+      if(colour == Colour.WHITE)
+      {
+         return whiteCastlingLong;
+      }
+      else
+      {
+         return blackCastlingLong;
+      }
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#setSquareAlPaso(es.dagaren.chesi.Square)
+    */
+   public void setEnPassantSquare(Square square)
+   {
+      if(square != this.enPassantSquare)
+      {
+         //Se actualiza la clave zobrist
+         zobristKey.xorEnPassant(enPassantSquare);
+         zobristKey.xorEnPassant(square);
+      }
+      this.enPassantSquare = square;
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#getSquareAlPaso()
+    */
+   public Square getEnPassantSquare()
+   {
+      return enPassantSquare;
+   }
+   
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#setNumMovement(int)
+    */
+   public void setMoveIndex(int moveIndex)
+   {
+      this.moveIndex = moveIndex;
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#getNumMovement()
+    */
+   public int getMoveIndex()
+   {
+      return moveIndex;
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#getNumJugada50()
+    */
+   public int getFiftyMovesRuleIndex()
+   {
+      return fiftyMovesRuleIndex;
+   }
+
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#setNumJugada50(int)
+    */
+   public void setFiftyMovesRuleIndex(int fiftyMovesRuleIndex)
+   {
+      this.fiftyMovesRuleIndex = fiftyMovesRuleIndex;
+   }
+   
+   
+   /**
+    * 
+    */
+   public boolean isCheckmate()
+   {
+      Colour turn = getTurn();
+      
+      //Para ver si es jaque mate primero se comprueba 
+      //si es jaque. Si no lo es se devuelve false
+      //inmediatamente
+      if(isInCheck(turn))
+      {
+         List<Movement> movesList = getMovements();
+      
+         for(Movement mov : movesList)
+         {
+            //Se realiza el movimiento
+            doMove(mov);
+            //Si tras realizar un movimiento no se encuentra en jaque
+            //quiere decir que al menos hay una jugada legal.
+            //Se deshace el movimiento y se devuelve false;
+            if(!isInCheck(turn))
+            {
+               undoMove(mov);
+               
+               return false;
+            }
+            
+            undoMove(mov);
+         }
+         
+         //Si tras probar todos los movimientos no se ha retornado
+         //quiere decir que no a habido ningún movimiento legal
+         //por lo que es jaque mate y se devuelve true
+         return true;
+      }
+      else
+      {
+         return false;
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 700:734, 743:777.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.